### DISCUSSION TITLE
any way to create dynamic enums for json schema from a database

### DISCUSSION THREAD

**Discussion by KaNe23** [pydantic/pydantic#2831] [Question]
Hello,
I have a fastAPI app with pydantic and I want to create a json schema, which is used to create a form by using:
https://github.com/json-editor/json-editor

The big problem I'm running into is, that drop down fields are defined via enums.
If I have a simple form:

```python
class Category(str, Enum):
    "Cat A" = "a"
    "Cat B" = "b"

class UserForm(BaseModel):
    categories: Optional[Category]
```
It generates a nice drop down menu for me:
```json
{
   "title":"UserForm",
   "type":"object",
   "properties":{
      "categories":{
         "$ref":"#/definitions/Category"
      }
   },
   "definitions":{
      "Category":{
         "title":"Category",
         "description":"An enumeration.",
         "enum":[
            "a",
            "b"
         ],
         "type":"string"
      }
   }
}
```

![image](https://user-images.githubusercontent.com/241655/119502708-80cc7c80-bd6a-11eb-94ad-104d3b8f55ee.png)

Now I want to feed those categories from the database.
But pydantic creates those schemas more or less statically from the typing(?).
I have tried so many things, but there seems to be no easy way for my setup to pipe the database session into a new class instance, that describes the schema and populate the enum.

G

**Comment by PrettyWood** [comment]
Hi @KaNe23 
Do you have a repo or code snippet to reproduce the error? Quite hard to help you without anything!

**Reply by KaNe23** [reply]
Hello @PrettyWood,
I was kinda worked out when I wrote that, looks really confusing.
I made a little example: https://github.com/KaNe23/FastAPI_Pydantic_Json_Schema

The first form is powered by a static enum model, which only lets you select something out of the drop down menu.
The second one is powered by the pydantic models, but creates an actual form to create categories.
![image](https://user-images.githubusercontent.com/241655/119678645-fb1b0080-be3f-11eb-87fb-71d0311dbbc9.png)

Now I would like to use the pydantic functionality to create a form for me, but I would like to add drop downs that are fed by the
database, in this example, I would want the categories from the database to be in the dropdown of the first form.
But I have no idea if that is actually possible.

I tried to actually instantiate a model and talk to the database to get the categories, but pydantic uses the typing system to
create those enums and I found no easy way, to manipulate the instantiated model enums somehow.

I hope that makes more sense. 

G
